source/models/bounties/Bounty.js

```diff
@@ -0,0 +1,124 @@
+﻿const { EmbedBuilder, Guild } = require('discord.js');
+const { DataTypes: { BIGINT, STRING, INTEGER, BOOLEAN }, Model } = require('sequelize');
+const { ihpAuthorPayload } = require('../../embedHelpers');
+const { Hunter } = require('../users/Hunter');
+const { Guild: HunterGuild } = require('../guilds/Guild');
+const { database } = require('../../../database');
+
+/** Bounties are user created objectives for other server members to complete */
+const bountyModel = {
+	id: {
+		primaryKey: true,
+		type: BIGINT,
+		autoIncrement: true
+	},
+	userId: {
+		type: STRING,
+		references: {
+			model: 'User',
+			key: 'id'
+		}
+	},
+	guildId: {
+		type: STRING,
+		references: {
+			model: 'Guild',
+			key: 'id'
+		}
+	},
+	postingId: {
+		type: STRING,
+	},
+	slotNumber: {
+		type: INTEGER,
+		allowNull: false
+	},
+	isEvergreen: {
+		type: BOOLEAN,
+		devaultValue: false
+	},
+	title: {
+		type: STRING,
+		allowNull: false,
+		defaultValue: "\u200B"
+	},
+	description: {
+		type: STRING,
+		allowNull: false,
+		defaultValue: "\u200B"
+	},
+	attachmentURL: {
+		type: STRING,
+		defaultValue: null
+	},
+	scheduledEventId: {
+		type: INTEGER,
+		defaultValue: null
+	},
+	state: { // Allowed values: "open", "completed", "deleted"
+		type: STRING,
+		defaultValue: "open"
+	},
+	completedAt: {
+		type: INTEGER,
+		defaultValue: null
+	},
+	deletedAt: {
+		type: INTEGER,
+		defaultValue: null
+	},
+	editCount: {
+		type: INTEGER,
+		defaultValue: 0
+	}
+};
+exports.Bounty = class Bounty extends Model {
+	/** Generate an embed for the given bounty
+	 * @param {Guild} guild
+	 * @param {Hunter?} hunter
+	 * @param {HunterGuild?} hunterGuild
+	 */
+	asEmbed(guild, hunter, hunterGuild) {
+		return guild.members.fetch(this.userId).then(async author => {
+			if (!hunter) {
+				hunter = await database.models.Hunter.findOne({ where: { userId: this.userId, guildId: this.guildId } });
+			}
+			if (!hunterGuild) {
+				hunterGuild = await database.models.Guild.findByPk(this.guildId);
+			}
+
+			const embed = new EmbedBuilder().setColor(author.displayColor)
+				.setAuthor(ihpAuthorPayload)
```

Comment:
This is likely due to me not being in the zone of discord.js for some time, but the `ihp` abbreviation feels a little too abstract, and I have no real idea of what this refers to. Is this `Imaginary Horizons Productions`? If so, that also feels too broad for this bot, and `bb` for `BountyBot` would make more sense in context.

Reply:
The constant is the data required to make the circled portion of an embed. I think the name is appropriate.
![Screenshot 2023-07-14 130427](https://github.com/Imaginary-Horizons-Productions/BountyBot/assets/6503151/ecebe91d-bce8-4685-8e24-fa2989844632)